lib/rules/arrow-spaces.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * @fileoverview Rule to require parens in arrow function arguments.
+ * @author Jxck
+ * @copyright 2015 Jxck. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    /**
+     * Merge default rules to given optional rules.
+     * @param {Object} options Optional rules
+     * @returns {Object} Merged rules
+     */
+    function mergeRule(options) {
+        // default
+        var rule = { before: true, after: true };
+
+        // merge args
+        if (options.length > 0) {
+            var option = options[0];
+
+            if (typeof option.before !== "undefined") {
+                rule.before = option.before;
+            }
+
+            if (typeof option.after !== "undefined") {
+                rule.after = option.after;
+            }
+        }
+
+        return rule;
+    }
+
+    /**
+     * Get tokens of arrow(`=>`) and before/after arrow.
+     * @param {ASTNode} node The arrow function node.
+     * @returns {Object} Tokens of arrow and before/after arrow.
+     */
+    function getTokens(node) {
+        var t = context.getFirstToken(node);
+        var before;
+        while (t.value !== "=>") {
+            before = t;
+            t = context.getTokenAfter(t);
+        }
+        var after = context.getTokenAfter(t);
+        return { before: before, arrow: t, after: after };
+    }
+
+    /**
+     * Count spaces before/after arrow(`=>`) token.
+     * @param {Object} tokens Tokens before/after arrow.
+     * @returns {Object} count of space before/after arrow.
+     */
+    function countSpaces(tokens) {
+        var before = tokens.arrow.range[0] - tokens.before.range[1];
+        var after = tokens.after.range[0] - tokens.arrow.range[1];
+        return { before: before, after: after };
+    }
+
+    /**
+     * Determines whether spaces before after arrow(`=>`) is satisfy rule.
+     * if before/after value is `true`, there should be one space.
+     * if before/after value is `false`, there should be no space.
+     * @param {ASTNode} node The arrow function node.
+     * @returns {void}
+     */
+    function spaces(node) {
+        var rule = mergeRule(context.options);
```

Comment:
I dont think this needs to be done every time this rule is called.
It should be done once during the initialization time.

Please look at other rules for example.